client/client_test.go

```diff
@@ -203,6 +203,27 @@ func (s *clientTestSuite) TestConn_Select() {
 	require.Equal(s.T(), "test1", e)
 }
 
+func (s *clientTestSuite) TestConn_Select2() {
+
+	str := `insert into test.mixer_test_conn (id, str, f, e, u, i,j) values (6, "a", 3.14, "test1", 255, -127,'{"key1": 1, "key2": null}'),(7, NULL,NULL, NULL, NULL, NULL,NULL)`
+	result, err := s.c.Execute(str)
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), uint64(2), result.AffectedRows)
+
+	str = `select * from mixer_test_conn where id in (6,7)`
+	ss := [][]string{{"6", "a", "3.14", "test1", "255", "-127", "{\"key1\": 1, \"key2\": null}"}, {"7", "", "", "", "", "", ""}}
```

Comment:
Thanks for adding test. LGTM now.

And I remember in https://github.com/go-mysql-org/go-mysql/issues/821#issuecomment-1722831730 you still need to wrap `'` to `String()` or use `NULL`. Do you think it's easier to use we implement these logic inside `String()`?

like

```golang
func (fv *FieldValue) String() string {
        if fv.Type == mysql.FieldValueTypeNull {
		return "NULL"
	}
	return "'" + string(fv.str) + "'"
}
```

Reply:
and we also check if the Type is number, we don't need to wrap `'`

Reply:
not so good, ```String()``` only needs to give the raw value. usually users will judge the ```NULL``` type and combine strings

like

```go
c.ExecuteSelectStreaming(mysql_select_all_query, &result, func(row []mysql.FieldValue) error {
	values := make([]string, 0, len(row))
	for _, v := range row {
		if v.Type == mysql.FieldValueTypeNull {
			values = append(values, "NULL")
		} else {
			values = append(values, fmt.Sprintf("'%s'", v.String()))
		}
	}
	r := "INSERT INTO xx VALUES (" + strings.Join(values, ",") + ")"
	fmt.Println(r)

	return nil
}, nil)
```

Reply:
insert into null value don't need to wrap '

Reply:
I think we don't need to insist "String() only needs to give the raw value". This is the comment of fmt.Stringer

```
// Stringer is implemented by any value that has a String method,
// which defines the “native” format for that value.
// The String method is used to print values passed as an operand
// to any format that accepts a string or to an unformatted printer
// such as Print.
```

I noticed that "String method is used to print values", so it better we directly let developer uses String() to build the SQL string rather than need to deal with NULL. As for what's "the 'native' format for that value", I think the String() should return a valid MySQL liternal https://dev.mysql.com/doc/refman/8.0/en/literals.html

Reply:
> insert into null value don't need to wrap '

I didn't say it need to wrap `'`.

Reply:
"String() only needs to give the raw value" raw value is ```LengthEncodedString(p[pos:])``` return value

Reply:
"String() only needs to give the raw value" is only your definition, I still prefer "String() should return a valid MySQL liternal https://dev.mysql.com/doc/refman/8.0/en/literals.html"

Reply:
using a different name?  just return the string value ```FieldValue.str```

Reply:
OK, and I think I can implement String() later, and then your application can directly use

```golang
	values := make([]string, 0, len(row))
	for _, v := range row {
		values = append(values, v.String())
	}
	r := "INSERT INTO xx VALUES (" + strings.Join(values, ",") + ")"
```

Maybe we can revert to

```golang
func (fv *FieldValue) Raw() []byte {
	return fv.str
}
```

Reply:
I guess I can implement String() in this or next weekend. Can you wait for me? If you are in hurry to support your application I can merge this PR after you change to `func (fv *FieldValue) Raw() []byte` function signature.

Reply:
thank! wait you.